iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -3600,6 +3631,11 @@ IOTHUB_CLIENT_RESULT IoTHubTransport_MQTT_Common_SetOption(TRANSPORT_LL_HANDLE h
             transport_data->auto_url_encode_decode = *((bool*)value);
             result = IOTHUB_CLIENT_OK;
         }
+        else if (strcmp(OPTION_TWIN_CONTENT_TYPE, option) == 0)
+        {
+            transport_data->twin_content_type = *((OPTION_TWIN_CONTENT_TYPE_VALUE*)value);
```

Comment:
validate the value is in the enum range?

Reply:
Assuming you mean the range of the explicit values of the enum members, and not within the available numeric range of the enum.  I could add a final enum member as an "OPTION_TWIN_CONTENT_TYPE_MAX_VALUE = 1" to check the value is equal or smaller than that.  It would need to be updated every time a new member is added to the enum.

Reply:
I guess you can enforce it in buildConfigForUsernameStep2IfNeeded(), but then you can't use the !=

Reply:
I think it's good to catch it at the time of the option being set.  I forgot to add error handling.  It return an invalid arg if out of range.